ecs-service/main.tf

```diff
@@ -0,0 +1,94 @@
+locals {
+  container_name        = var.container_name != "" ? var.container_name : var.container_definitions[0].name
+  container_definitions = jsonencode([for definition in module.container_definition : definition.json_map_object])
+
+  launch_type = "FARGATE"
+  requires_compatibilities = {
+    FARGATE : ["FARGATE"]
+  }
+  network_mode = "awsvpc"
+  target_type  = local.network_mode == "awsvpc" ? "ip" : "instance"
```

Comment:
eh? :)

Reply:
:)
AWSVPC is required for Fargate and IP is required for AWSVPC network mode. 

> instance -  The targets are specified by instance ID.
> ip - The targets are IP addresses.
> lambda - The target is a Lambda function.

https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-target-groups.html#target-type

Reply:
instance means ec2 instance here?

Reply:
Yes. It would be instance for EC2 or even for docker running on EC2 backed ECS cluster.